include/power/pm_policy.h

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_POWER_PM_POLICY_H_
+#define ZEPHYR_INCLUDE_POWER_PM_POLICY_H_
+
+#include <power/power.h>
+
+#include <string.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief PM Policy definition
+ *
+ * Store information about a power management policy.
+ */
+struct pm_policy {
+	int (*next_state_get)(struct pm_policy *policy,
+			      int32_t ticks, struct pm_state_info *info);
+	int (*low_power_devices_get)(struct pm_policy *policy,
+				     enum pm_state state);
+};
+
+/**
+ * @brief Statically define and register a power management policy
+ *
+ * @param policy_name The name of this policy
+ * @param policy_next_state_get Callback to get the next PM state
+ * @param policy_low_power_devices_get Callback to determine whether to put
+ *        or not devices in low power state, given the system PM state
+ */
+#define PM_POLICY_DEFINE(policy_name, policy_next_state_get,		\
+			 policy_low_power_devices_get)			\
+	const struct pm_policy policy_name = {				\
+		.next_state_get = policy_next_state_get,		\
+		.low_power_devices_get = policy_low_power_devices_get,	\
+	};								\
+	const struct pm_policy *z_sys_pm_policy = &policy_name
+
+/**
+ * @brief Function to create device PM list
+ */
+void pm_create_device_list(void);
+
+/**
+ * @brief Function to suspend the devices in PM device list
+ */
+int pm_suspend_devices(void);
+
+/**
+ * @brief Function to put the devices in PM device list in low power state
+ */
+int pm_low_power_devices(void);
+
+/**
+ * @brief Function to force suspend the devices in PM device list
+ */
+int pm_force_suspend_devices(void);
+
+/**
+ * @brief Function to resume the devices in PM device list
+ */
+void pm_resume_devices(void);
+
+/**
+ * @brief Function to get the next PM state based on the ticks
+ *
+ * @return 0 in case of success, a negative value otherwise
```

Comment:
Please document the known values.  It's fine to have a fallback for future-proofing.  E.g.

```
 * @return 
 *   * 0 in case of success
 *   * -ENOTSUP if there is no policy, or the police dpes not support a next-state capability;
 *   * other negative error codes indicating specific failure modes
```